lib/spack/spack/modules.py

```diff
@@ -388,6 +388,7 @@ def write(self, overwrite=False):
             for mod in modules:
                 set_module_variables_for_package(package, mod)
             set_module_variables_for_package(package, package.module)
+            package.setup_environment(spack_env, env)
```

Comment:
@citibeth This is part of the issues you are seeing in #3059 . Trying to fix the other part (if you look into your modules the unwanted variables are likely to be present multiple times).